front/temporal/usage_queue/client.ts

```diff
@@ -0,0 +1,66 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok, rateLimiter } from "@dust-tt/types";
+
+import { getTemporalClient } from "@app/lib/temporal";
+import logger from "@app/logger/logger";
+import { QUEUE_NAME } from "@app/temporal/usage_queue/config";
+import { updateWorkspaceUsageWorkflow } from "@app/temporal/usage_queue/workflows";
+
+async function shouldProcessUsageUpdate(workflowId: string) {
+  // Compute the max usage of the workspace once per hour.
+  const hasRunInPastHour = await rateLimiter({
```

Comment:
Since this is in the critical path (called at every user message), did you double-check this call was very fast?

Reply:
Yes from my test, reading a value in Redis is more efficient than starting a new Temporal workflow.